CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconsRenderable.java

```diff
@@ -462,218 +758,330 @@ private void DestroyVertexBuffers() {
+                                                             VertexUniformBufferObject.SIZEOF,
+                                                             VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+                                                             VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
+            vertexUniformBuffer.DEBUGNAME = String.format("CVKIconsRenderable vertexUniformBuffer %d", i);   
+            vertexUniformBuffers.add(vertexUniformBuffer);                     
+        }        
+        return UpdateVertexUniformBuffers(stack);
+    }
+        
+    private int UpdateVertexUniformBuffers(MemoryStack stack) {
         CVKAssert(cvkSwapChain != null);
+        CVKAssert(cvkVertexUBStagingBuffer != null);
+        CVKAssert(vertexUniformBuffers != null);
+        CVKAssert(vertexUniformBuffers.size() > 0);
+        
+        int ret = VK_SUCCESS;
         
-        int imageCount = cvkSwapChain.GetImageCount();        
-
-//        vertexUniformBuffers = new ArrayList<>();
-//        geometryUniformBuffers = new ArrayList<>();        
-//        for (int i = 0; i < imageCount; ++i) {   
-//            CVKBuffer vertUniformBuffer = CVKBuffer.Create(cvkDevice, 
-//                                                          CVKFPSRenderable.SIZEOF,
-//                                                          VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
-//                                                          VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
-//            vertUniformBuffer.DEBUGNAME = String.format("CVKFPSRenderable vertUniformBuffer %d", i);   
-//            vertexUniformBuffers.add(vertUniformBuffer);            
-//            
-//            CVKBuffer geomUniformBuffer = CVKBuffer.Create(cvkDevice, 
-//                                                          CVKFPSRenderable.SIZEOF,
-//                                                          VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
-//                                                          VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
-//            geomUniformBuffer.DEBUGNAME = String.format("CVKFPSRenderable geomUniformBuffer %d", i);                                    
-//            geometryUniformBuffers.add(geomUniformBuffer);            
-//        }
-        
-        return UpdateUniformBuffers(stack);                
+        // Populate the UBO.  This is easy to deal with, but not super efficient
+        // as we are effectively staging into the staging buffer below.
+        vertexUBO.mvMatrix = parent.getDisplayModelViewMatrix();
+        vertexUBO.morphMix = parent.getDisplayCamera().getMix();
+        
+        // TODO: replace with constants.  In the JOGL version these were in a static var CAMERA that never changed
+        vertexUBO.visibilityLow = parent.getDisplayCamera().getVisibilityLow();
```

Comment:
Note: If these are constants, then the vertex shader could move over to being pushconstants.

Reply:
Can you please add a task for this?  If possible link it to #20 as that will reduce some size in that UBO.

Reply:
Added task